registry/registry.go

```diff
@@ -133,9 +114,11 @@ func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
 // DockerHeaders returns request modifiers that ensure requests have
 // the User-Agent header set to dockerUserAgent and that metaHeaders
 // are added.
-func DockerHeaders(metaHeaders http.Header) []transport.RequestModifier {
+func DockerHeaders(userAgent string, metaHeaders http.Header) []transport.RequestModifier {
 	modifiers := []transport.RequestModifier{
-		transport.NewHeaderRequestModifier(http.Header{"User-Agent": []string{dockerUserAgent}}),
+		transport.NewHeaderRequestModifier(http.Header{
+			"User-Agent": []string{userAgent},
```

Comment:
True, I actually think `userAgent == ""` should be an error, since we want to enforce that requests have a proper user agent.

If you agree, I will make the change.